server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesCreationProcessor.java

```diff
@@ -19,20 +19,97 @@
+    @VisibleForTesting
+    SetMailboxesCreationProcessor(MailboxManager mailboxManager, MailboxUtils<Id> mailboxUtils) {
+        this.mailboxManager = mailboxManager;
+        this.mailboxHierarchySorter =
+            new MailboxHierarchySorter<>(
+                x -> x.getKey().getCreationId(),
+                x -> x.getValue().getParentId());
+        this.mailboxUtils = mailboxUtils;
+    }
+
+    public SetMailboxesResponse process(SetMailboxesRequest request, MailboxSession mailboxSession) {
         SetMailboxesResponse.Builder builder = SetMailboxesResponse.builder();
-        request.getCreate().entrySet().stream()
-            .forEach(entry -> builder.creation(
-                entry.getKey(),
-                Mailbox.builder().name(entry.getValue().getName()).id("serverId").build()));
+        Map<MailboxCreationId, String> creationIdsToCreatedMailboxId = new HashMap<>();
+        mailboxHierarchySorter.sortFromRootToLeaf(request.getCreate().entrySet())
+            .forEach(entry -> {
+                createMailbox(entry.getKey(), entry.getValue(), mailboxSession, creationIdsToCreatedMailboxId, builder);
+            });
         return builder.build();
     }
 
+    private void createMailbox(MailboxCreationId mailboxCreationId, MailboxRequest mailboxRequest, MailboxSession mailboxSession,
+            Map<MailboxCreationId, String> creationIdsToCreatedMailboxId, SetMailboxesResponse.Builder builder) {
+        try {
+            MailboxPath mailboxPath = getMailboxPath(mailboxRequest, creationIdsToCreatedMailboxId, mailboxSession);
+            mailboxManager.createMailbox(mailboxPath, mailboxSession);
+            Optional<Mailbox> mailbox = mailboxUtils.mailboxFromMailboxPath(mailboxPath, mailboxSession);
+            if (mailbox.isPresent()) {
+                builder.creation(mailboxCreationId, mailbox.get());
+                creationIdsToCreatedMailboxId.put(mailboxCreationId, mailbox.get().getId());
+            } else {
+                builder.notCreated(mailboxCreationId, SetError.builder()
+                        .type("anErrorOccurred")
+                        .description("An error occurred when creating the mailbox")
+                        .build());
+            }
+        } catch (MailboxParentNotFoundException e) {
+            builder.notCreated(mailboxCreationId, SetError.builder()
+                    .type("invalidArguments")
+                    .description("The parent mailbox was not found.")
+                    .build());
+        } catch (MailboxException e) {
+            builder.notCreated(mailboxCreationId, SetError.builder()
+                    .type("anErrorOccurred")
+                    .description("An error occurred when creating the mailbox")
+                    .build());
+        }
+    }
+
+    private MailboxPath getMailboxPath(MailboxRequest mailboxRequest, Map<MailboxCreationId, String> creationIdsToCreatedMailboxId, MailboxSession mailboxSession) throws MailboxException {
```

Comment:
getMailboxNameFromId throws because getMailboxFromId (which has only 1 reference) throws.
Only consuming method has lambda and has, as a consequence, to deal with throwing functions
So it's pretty easy to remove all the throws by only catching in getMailboxFromId and returning empty (valid here because this is a getter).